src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,98 @@
+package duke.task;
+
+import java.time.format.DateTimeParseException;
+
+import duke.util.DateTime;
+import duke.util.DukeException;
+
+/**
+ * Event is created by 'event eat breakfast /at 0800'.
+ * Events are a type of Task.
+ */
+public class Event extends Task {
+    public static final char SYMBOL = 'E';
+
+    private static final String EVENT_DELIMITER = "/at";
+
+    // Messages
+    private static final String INVALID_EVENT_MESSAGE = "Invalid use of event command."
+            + " Use 'event <text> /at <datetime>'";
+    private static final String INVALID_EVENT_FORMAT_MESSAGE = "Invalid DateTime format for event. "
+            + "DateTime must be in the format of yyyy-MM-dd HHmm (2019-02-01 1800)";
+    private static final String MISSING_EVENT_MESSAGE = "Some arguments are missing. Use 'event <text> /at <datetime>'";
+    private static final String INVALID_SAVE_MESSAGE = "Event save is given in the wrong format";
+
+    private final DateTime eventDatetime;
+
+    /**
+     * Constructor for the event object.
+     *
+     * @param text The description of the event
+     * @param eventDatetime The date where it is due, to be converted to a DateTime object
+     * @param isDone Whether the Event is finished
+     */
+    private Event(String text, String eventDatetime, boolean isDone) throws DukeException {
+        super(text, isDone);
+        try {
+            this.eventDatetime = new DateTime(eventDatetime);
+        } catch (DateTimeParseException err) {
+            throw new DukeException(INVALID_EVENT_FORMAT_MESSAGE);
+        }
+    }
+
+    /**
+     * Factory method for creating an event object.
+     * @param input The remaining input after the initial 'event' string
+     * @param isDone Whether the Event is finished
+     * @return An Event object
+     * @throws DukeException An exception thrown according to the message given
+     */
+    public static Event newEvent(String input, boolean isDone) throws DukeException {
```

Comment:
Perhaps it would be better to change this to createNewEvent to change this to a verb?